src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/ShellInProcess.cs

```diff
@@ -8,17 +8,65 @@
 using System.Reflection;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Windows.Controls;
+using System.Windows.Input;
 using Microsoft.CodeAnalysis.UnitTests;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Threading;
+using Xunit;
 using IAsyncDisposable = System.IAsyncDisposable;
 
 namespace Microsoft.VisualStudio.Extensibility.Testing
 {
     internal partial class ShellInProcess
     {
+        /// <returns>True if the AllInOneSearch is being used for Navigation</returns>
+        public async Task<bool> ShowNavigateToDialogAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            await TestServices.Shell.ExecuteCommandAsync(VSConstants.VSStd12CmdID.NavigateTo, cancellationToken);
+
+            return await WaitForNavigateToFocusAsync(cancellationToken);
```

Comment:
Why have the local function here?

Reply:
Sometimes I like to name a bit of code to make it very clear what the code is trying to achieve. Of course I could do that with a comment as well.